### PR TITLE
Ab#600 narrative panel color room name and direction

### PR DESCRIPTION
**Pull request by wang-xiaotian**

### CODE REVIEW

**Comment by chaynsaw** [comment]
I think there is a bug here, can you confirm? When I click away from the main window and try to enter a command, I am not allowed to do so until I press "tab".

**Comment by wang-xiaotian** [comment]
@chaynsaw that was a bug previously. i already resolved that. i will merge conflicts first and you can try again

**Comment by wang-xiaotian** [comment]
@chaynsaw what happened was when you clicked my map panel, the console panel lost focus from the window. I disabled focus on the map panel. it should be good now

File: src/main/java/com/locallampoon/fiveh/core/Player.java [hunk 1]
```diff
@@ -216,10 +218,10 @@ void flee(Map<String, Room> houseMap) {
 
     void getCurrentItemDetails() {
```
Comment (chaynsaw) [inline src/main/java/com/locallampoon/fiveh/core/Player.java:219]: Nothing to do with you, but I just noticed that this never gets triggered. All inventories start as a 5-length array of "", so they are never empty.

File: src/main/java/com/locallampoon/fiveh/ui/NarrativePanel.java [hunk 2]
```diff
@@ -33,8 +32,41 @@ public void setTextArea(String text) {
...
+     */
+    public void appendTextArea(String text, Color color)
+    {
+        this.enableNarrative();
+        this.appendToPane(this.textArea, text, color);
+        this.disableNarrative();
+    }
+
+    public void disableNarrative(){
+        textArea.setEditable(false);
```
Comment (chaynsaw) [inline src/main/java/com/locallampoon/fiveh/ui/NarrativePanel.java:56]: Can you explain again what is the purpose of using setEditable here?

File: src/main/java/com/locallampoon/fiveh/core/Player.java (hunk 1 above)
Comment (wang-xiaotian) [reply src/main/java/com/locallampoon/fiveh/core/Player.java:219]: Ok. few functions in our code never got triggered. If it is created by me, i will delete it

File: src/main/java/com/locallampoon/fiveh/ui/NarrativePanel.java [hunk 3]
```diff
@@ -33,8 +32,41 @@ public void setTextArea(String text) {
...
+     * @param color
+     */
+    public void appendTextArea(String text, Color color)
+    {
+        this.enableNarrative();
+        this.appendToPane(this.textArea, text, color);
+        this.disableNarrative();
+    }
+
+    public void disableNarrative(){
```
Comment (chaynsaw) [inline src/main/java/com/locallampoon/fiveh/ui/NarrativePanel.java]: Minor detail, but I think you should rename this method to `disableNarrativeText`... and also the other to `enableNarrativeText`. It is more specific and less confusing than the existing way.

File: src/main/java/com/locallampoon/fiveh/ui/NarrativePanel.java (hunk 2 above)
Comment (wang-xiaotian) [reply src/main/java/com/locallampoon/fiveh/ui/NarrativePanel.java:56]: when i setEditable(false), it is still isEnabled() = true, but it won't allow user to edit the content inside textArea

File: src/main/java/com/locallampoon/fiveh/ui/NarrativePanel.java (hunk 3 above)
Comment (wang-xiaotian) [reply src/main/java/com/locallampoon/fiveh/ui/NarrativePanel.java]: this is totally fine. i will make changes

File: src/main/java/com/locallampoon/fiveh/ui/NarrativePanel.java (hunk 2 above)
Comment (wang-xiaotian) [reply src/main/java/com/locallampoon/fiveh/ui/NarrativePanel.java:56]: if it is not editable, it blocks all color changes

File: src/main/java/com/locallampoon/fiveh/ui/NarrativePanel.java (hunk 3 above)
Comment (wang-xiaotian) [reply src/main/java/com/locallampoon/fiveh/ui/NarrativePanel.java]: I will do that tomorrow morning ... :)

**Review by chaynsaw — APPROVED** [review]

**Pull request merged**